### PR TITLE
[squeezebox] Add new channels for additional tags

### LINKED ISSUE #13353: [squeezebox] Add channels for composer, band and conductor tags.
In the squeezebox add-in, there is only a limited set of tags of the played music piece, such as title, artist, year and so on available as player thing channels.

It would be great to enhance this set by some tags commonly used with classical music pieces, such as:

COMPOSER
BAND
CONDUCTOR

Is it possible to add those?

### PR DESCRIPTION
**Pull request by mhilbush**
Resolves #13353 

Added channels for the following:
- albumartist
- trackartist
- band
- conductor
- composer

Signed-off-by: Mark Hilbush <mark@hilbush.com>

### CODE REVIEW

File: bundles/org.openhab.binding.squeezebox/README.md [hunk 1]
```diff
@@ -115,6 +115,11 @@ All devices support some of the following channels:
 | artist                 | String    | Artist name of the current song                                                              |
 | year                   | String    | Release year of the current song                                                             |
 | genre                  | String    | Genre name of the current song                                                               |
+| albumartist            | String    | Main artist of the entire album                                                              |
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/README.md:118]: Please adopt camelCase like for most other existing channels.
```suggestion
| albumArtist            | String    | Main artist of the entire album                                                              |
```

File: bundles/org.openhab.binding.squeezebox/README.md [hunk 2]
```diff
@@ -115,6 +115,11 @@ All devices support some of the following channels:
...
+| trackartist            | String    | Main artist of the track                                                                     |
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/README.md:119]:
```suggestion
| trackArtist            | String    | Main artist of the track                                                                     |
```

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxBindingConstants.java [hunk 3]
```diff
@@ -59,6 +59,11 @@ public class SqueezeBoxBindingConstants {
     public static final String CHANNEL_ARTIST = "artist";
     public static final String CHANNEL_YEAR = "year";
     public static final String CHANNEL_GENRE = "genre";
+    public static final String CHANNEL_ALBUMARTIST = "albumartist";
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxBindingConstants.java:62]:
```suggestion
    public static final String CHANNEL_ALBUMARTIST = "albumArtist";
```

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxBindingConstants.java [hunk 4]
```diff
@@ -59,6 +59,11 @@ public class SqueezeBoxBindingConstants {
...
+    public static final String CHANNEL_TRACKARTIST = "trackartist";
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxBindingConstants.java:63]:
```suggestion
    public static final String CHANNEL_TRACKARTIST = "trackArtist";
```

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxPlayerEventListener.java [hunk 5]
```diff
@@ -73,6 +73,16 @@ public interface SqueezeBoxPlayerEventListener {
 
     void genreChangeEvent(String mac, String genre);
 
+    void albumartistChangeEvent(String mac, String albumartist);
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxPlayerEventListener.java:76]:
```suggestion
    void albumArtistChangeEvent(String mac, String albumartist);
```

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxPlayerEventListener.java [hunk 6]
```diff
@@ -73,6 +73,16 @@ public interface SqueezeBoxPlayerEventListener {
...
+
+    void trackartistChangeEvent(String mac, String trackartist);
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxPlayerEventListener.java:78]:
```suggestion
    void trackArtistChangeEvent(String mac, String trackartist);
```

File: bundles/org.openhab.binding.squeezebox/src/main/resources/OH-INF/thing/thing-types.xml [hunk 7]
```diff
@@ -79,6 +79,11 @@
 			<channel id="artist" typeId="artist"/>
 			<channel id="year" typeId="year"/>
 			<channel id="genre" typeId="genre"/>
+			<channel id="albumartist" typeId="albumartist"/>
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/src/main/resources/OH-INF/thing/thing-types.xml:82]:
```suggestion
			<channel id="albumArtist" typeId="albumArtist"/>
```

File: bundles/org.openhab.binding.squeezebox/src/main/resources/OH-INF/thing/thing-types.xml [hunk 8]
```diff
@@ -79,6 +79,11 @@
...
+			<channel id="trackartist" typeId="trackartist"/>
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/src/main/resources/OH-INF/thing/thing-types.xml:83]:
```suggestion
			<channel id="trackArtist" typeId="trackArtist"/>
```

File: bundles/org.openhab.binding.squeezebox/src/main/resources/OH-INF/thing/thing-types.xml [hunk 9]
```diff
@@ -282,6 +287,36 @@
 		<description>Genre name of the current song</description>
 		<state readOnly="true" pattern="%s"></state>
 	</channel-type>
+	<channel-type id="albumartist" advanced="true">
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/src/main/resources/OH-INF/thing/thing-types.xml:290]:
```suggestion
	<channel-type id="albumArtist" advanced="true">
```

File: bundles/org.openhab.binding.squeezebox/src/main/resources/OH-INF/thing/thing-types.xml [hunk 10]
```diff
@@ -282,6 +287,36 @@
...
+		<item-type>String</item-type>
+		<label>Album Artist</label>
+		<description>Main artist for the album</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
+	<channel-type id="trackartist" advanced="true">
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/src/main/resources/OH-INF/thing/thing-types.xml:296]:
```suggestion
	<channel-type id="trackArtist" advanced="true">
```

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxServerHandler.java [hunk 11]
```diff
@@ -674,7 +674,8 @@ private void handleClientMessage(final String mac, String[] messageParts) {
         }
 
         private void handleStatusMessage(final String mac, String[] messageParts) {
-            String remoteTitle = "", artist = "", album = "", genre = "", year = "";
+            String remoteTitle = "", artist = "", album = "", genre = "", year = "", albumartist = "", trackartist = "",
```
Comment (jlaur) [inline bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxServerHandler.java:677]:
```suggestion
            String remoteTitle = "", artist = "", album = "", genre = "", year = "", albumArtist = "", trackArtist = "",
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thank you. Looks simple and good to me! I have added comments for a request to use camelCase for the two multi-worded new channels.

**Comment by mhilbush** [comment]
Ha ha. I actually started down the camelCase path. But I changed it to be consistent with the tag naming convention in the LMS, and also because I didn't want to change remotetitle and coverartdata to camelCase, as that would be a breaking change.

I admit that the binding is a bit inconsistent in how the channels are named. Some are camelCase and some are not. 

But I'm happy to change the new channels to camelCase if that's the recommendation. Just let me know.

**Comment by jlaur** [comment]
> Ha ha. I actually started down the camelCase path. But I changed it to be consistent with the tag naming convention in the LMS, and also because I didn't want to change remotetitle and coverartdata to camelCase, as that would be a breaking change.

Correct, it's a pickle staying consistent and conforming to naming conventions and at the same time without breaking backwards compatibility. 🙂 However, one thing we shouldn't do is adopting naming conventions from the API's we implement in bindings. We have our own, making openHAB consistent across bindings and being API agnostic.

> I admit that the binding is a bit inconsistent in how the channels are named. Some are camelCase and some are not.

For consistency we will have to choose one of the two (camelCase or lowercase). See below.

> But I'm happy to change the new channels to camelCase if that's the recommendation. Just let me know.

Yes, that's the recommendation. And since it's already in use for some of the channels, this would be the preferred way going forward. See also https://community.openhab.org/t/naming-convention-for-channels/128841/10.

Had all the existing channel been lowercased (or underscore_separated or dash-separated), the situation would be different. Here I would recommend to prioritize consistency higher than the current recommendation.

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**